guides/manipulating-pages.rst

```diff
@@ -1,102 +1,96 @@
 Manipulating Pages
 ==================
 
-Ok, you've got an interesting page element. Now you'll want to do something
-with it. ``Behat\Mink\Element\NodeElement`` provides bunch a of useful methods
-for you:
+Once you :doc:`get a page element </guides/traversing-pages>`, you will want
+to manipulate it. You can also interact with the page, which is covered in
+the :doc:`next chapter </guides/interacting-with-pages>`.
+
+Getting the tag name
+--------------------
+
+The ``NodeElement::getTagName`` method allows you to get the tag name of
+the element. This tag name is always returned lowercased.
 
 .. code-block:: php
 
     $el = $page->find('css', '.something');
 
     // get tag name:
-    echo $el->getTagName();
+    echo $el->getTagName(); // displays 'a'
 
-    // check that element has href attribute:
-    $el->hasAttribute('href');
+Accessing HTML attributes
+-------------------------
 
-    // get element's href attribute:
-    echo $el->getAttribute('href');
-
-Element Content and Text
-~~~~~~~~~~~~~~~~~~~~~~~~
-
-To retrieve HTML content or plain text from out of the element, you can use:
-
-.. code-block:: php
+The ``NodeElement`` class gives you access to HTML attributes of the element.
 
-    $plainText = $el->getText();
-    $html = $el->getHtml();
-
-.. note::
-
-    ``getText()`` will strip tags and unprinted characters out of the response,
-    including newlines. So it'll basically return the text, that user sees
-    on the page.
+``NodeElement::hasAttribute``
+    Checks whether the element has a given attribute.
 
-Form Field Manipulations
-~~~~~~~~~~~~~~~~~~~~~~~~
+``NodeElement::getAttribute``
+    Gets the value of an attribute.
 
-You can fill form fields/retrieve its values with form manipulation actions:
+``NodeElement::hasClass``
+    Checks whether the element has the given class (convenience wrapper around
+    ``getAttribute('class')``).
 
 .. code-block:: php
 
-    // check/uncheck checkbox:
-    if ($el->isChecked()) {
-        $el->uncheck();
-    }
-    $el->check();
-
-    // select option in select:
-    $el->selectOption('option value');
+    $el = $page->find('css', '.something');
 
-    // attach file to file input:
-    $el->attachFile('/path/to/file');
+    if ($el->hasAttribute('href') {
+        echo $el->getAttribute('href');
+    } else {
+        echo 'This anchor is not a link. It does not have an href.';
+    }
 
-    // get input value:
-    echo $el->getValue();
+Element Content and Text
+------------------------
 
-    // set intput value:
-    $el->setValue('some val');
+The ``Element`` class provides access to the content of elements.
 
-    // press the button:
-    $el->press();
+``Element::getHtml``
+    Gets the inner HTML of the element, i.e. all children of the element.
 
+``Element::getOuterHtml``
+    Gets the outer HTML of the element, i.e. including the element itself.
 
-Mouse Manipulations
-~~~~~~~~~~~~~~~~~~~
+``Element::getText``
+    Gets the text of the element.
```

Comment:
In general this is correct, but in reality we're also getting text content from tags inside the tag we're operating on. For example in this HTML:

```html
<div id="test">
this
<p> is correct </p>
behavior
```
text content for DIV will be `this is correct behavior` and not `this behavior`.